tests/test_qnode.py

```diff
@@ -1597,6 +1550,162 @@ def circuit(x, y):
             assert info["device_name"] == "default.qubit.autograd"
 
 
+# pylint: disable=unused-argument
+class CustomDevice(experimental.Device):
+    """A null device that just returns 0."""
+
+    def __repr__(self):
+        return "CustomDevice"
+
+    def execute(self, circuits, execution_config=None):
```

Comment:
Why it does not work with execute?

Reply:
What do you mean?

Reply:
This is just a local custom device so that we are more so testing against the interface, rather then one particular implementation.